common/flatpak-dir.c

```diff
@@ -7793,13 +7855,75 @@ flatpak_dir_install (FlatpakDir          *self,
         }
       else
         {
-          /* We're pulling from a remote source, we do the network mirroring pull as a
-             user and hand back the resulting data to the system-helper, that trusts us
-             due to the GPG signatures in the repo */
+          g_autoptr (GError) local_error = NULL;
+          g_autofree gchar *srcdir = NULL;
+          g_autofree gchar *mountpoint = NULL;
+          gint socket = -1;
+
+          if (!flatpak_dir_system_helper_call_get_revokefs_fd (self,
+                                                               0,
+                                                               ref,
+                                                               opt_commit,
+                                                               installation ? installation : "",
+                                                               &socket,
+                                                               &srcdir,
+                                                               &mountpoint,
+                                                               cancellable,
+                                                               &local_error))
+            {
+              if (g_error_matches (local_error, G_DBUS_ERROR, G_DBUS_ERROR_NOT_SUPPORTED))
+                g_debug ("revokefs-fuse not supported on your installation: %s", local_error->message);
+              else
+                g_warning ("Failed to get revokefs socket from system-helper: %s", local_error->message);
+            }
+          else
+            {
+              g_autoptr(GSubprocess) revokefs_fuse = NULL;
+              g_autoptr(GSubprocessLauncher) launcher = NULL;
+              g_autofree gchar *client_uid = NULL;
+              g_autofree gchar *revokefs_fuse_path = NULL;
+
+              revokefs_fuse_path = g_find_program_in_path ("revokefs-fuse");
+
+              client_uid = g_strdup_printf ("uid=%d", getuid ());
+              launcher = g_subprocess_launcher_new (G_SUBPROCESS_FLAGS_NONE);
+              g_subprocess_launcher_take_fd (launcher, socket, 3);
+              revokefs_fuse  = g_subprocess_launcher_spawn (launcher,
+                                                            &local_error,
+                                                            revokefs_fuse_path, "-o", client_uid, "-s", "--socket=3",
+                                                            srcdir, mountpoint, NULL);
+              if (local_error != NULL)
+                {
+                  g_warning ("Error spawning revokefs: %s", local_error->message);
+                  g_clear_error (&local_error);
+                }
+
+              if (revokefs_fuse &&
+                  g_subprocess_wait (revokefs_fuse, NULL, &local_error) &&
+                  g_subprocess_get_exit_status (revokefs_fuse) == 0)
```

Comment:
You could replace `wait` + `get_exit_status() == 0` with `g_subprocess_wait_check()`.